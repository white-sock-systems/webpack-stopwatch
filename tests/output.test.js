const webpack = require('webpack');
const Plugin = require('../index');
const path = require('path');
const fs = require('fs');

const co = (compiler) =>
    new Promise((resolve, reject) => {
        compiler.run((error, stats) => {
            if (error) {
                return reject(error);
            }

            return resolve(stats);
        });
    });

const compiler = webpack({
    mode: 'development',
    entry: './sample-application/src/index.js',
    plugins: [new Plugin({ statsFolder: '.stats', quiet: true })],
    output: {
        filename: 'main.js',
        path: path.resolve(__dirname, 'dist'),
    },
});

describe('Compile Output', () => {
    test('should match previous output', async () => {
        const res = await co(compiler);
        // hash = content of output
        expect(res.compilation.fullHash).toMatchSnapshot();
    });
});
