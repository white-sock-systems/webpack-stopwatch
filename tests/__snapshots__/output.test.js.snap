// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compile Output should pass options to plugin 1`] = `"55d43834682a19b3b501bbc22ccb4c32"`;
